pandas/core/indexes/datetimelike.py

```diff
@@ -163,6 +164,20 @@ def values(self):
     def asi8(self):
         return self._data.asi8
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
sidenote (not for this PR): we should move to `__array_ufunc__` here instead

Reply:
agreed on this, can you create an issue @jbrockmendel